docs/guide/arrays.md

```diff
@@ -0,0 +1,79 @@
+# Arrays
+
+HyperFormula lets you perform operations on arrays.
+
+You can:
+- Operate on arrays just like on non-array values
+- Pass arrays (instead of values) to functions
+- Constrain the output array's size
+- Filter an array based on boolean arrays
+
+## About arrays in HyperFormula
+
+There are two types of arrays in HyperFormula:
+* **Range array**: a range of cell addresses (e.g. `A1:A10`)
+* **Ad-hoc array**: an array with no ground-truth range that defines the values (e.g. `5*A1:B5`)
+
+Non-array values are **scalars**.
+
+By default, array arithmetic is disabled globally in HyperFormula.
+
+To use array arithmetic, either enable it globally ([set the `useArrayArithmetic` option to `true`](../api/interfaces/configparams.html#usearrayarithmetic)), or use the ARRAYFORMULA function.
+
+## Operating on arrays
+
+When you perform an operation on an array, each output array value is a result of that operation on the corresponding input array value.
+
+To enable array arithmetic for your formula, use the [`ARRAYFORMULA`](built-in-functions.md#array-manipulation) function:
+
+| Syntax | Example |
+| :--- | :--- |
+| `ARRAYFORMULA(your_formula)` | `=ARRAYFORMULA(A2:A5*B2:B5)` |
+
+If you don't use the `ARRAYFORMULA` function, HyperFormula treats [ad-hoc arrays](#about-arrays-in-hyperformula) as [scalars](#about-arrays-in-hyperformula), taking only your ad-hoc array's top-left value.
+
+To enable array arithmetic globally, set the `useArrayArithmetic` option to `true` in HyperFormula [configuration](../api/interfaces/configparams.html#usearrayarithmetic).
+
+## Passing arrays to functions
+
+When you pass an array as a function's argument, the function returns an array as well.
+
+To pass an array as an argument, enable the array arithmetic with the [`ARRAYFORMULA`](built-in-functions.md#array-manipulation) function:
+
+| Syntax | Example |
+| :--- | :--- |
+| `YOUR_FUNCTION(ARRAYFORMULA(your_formula))` | `=ISEVEN(ARRAYFORMULA(A2:A5*10))` |
```

Comment:
This example is wrong, this explanation is wrong.

ARRAYFORMULA would enable `your_formula` to use array arithmetic. But `YOUR_FUNCTION` is not in array arithmetic mode, so it would cast array to scalar. Proper way would be:

```suggestion
| `ARRAYFORMULA(YOUR_FUNCTION(your_formula))` | `=ARRAYFORMULA(ISEVEN(A2:A5*10))` |
```